src/handler.js

```diff
@@ -148,6 +148,7 @@ function ssr(req) {
  */
 function get_origin(headers) {
   const protocol = (protocol_header && headers.get(protocol_header)) || "https";
+  host_header = host_header || "host";
```

Comment:
I notice you're providing a default value for `host_header`. Consider utilizing the `env` function's fallback parameter on line 23 to set "Host" as the fallback. For example: `const host_header = env("HOST_HEADER", "Host").toLowerCase();`
This approach keeps default configurations centralized and also aligns with the common practice of capitalizing standard HTTP header names.